chia/full_node/full_node.py

```diff
@@ -1146,8 +1195,12 @@ async def fetch_blocks(output_queue: asyncio.Queue[Optional[tuple[WSChiaConnecti
                         self.log.error(f"failed fetching {start_height} to {end_height} from peers")
                         return
                     if self.sync_store.peers_changed.is_set():
-                        new_peers_with_peak = self.get_peers_with_peak(peak_hash)
+                        existing_peers = {c: timestamp for c, timestamp in new_peers_with_peak}
+                        peers = self.get_peers_with_peak(peak_hash)
+                        new_peers_with_peak = [(c, existing_peers.get(c, end)) for c in peers]
```

Comment:
Not making the connection hashable avoids us needing to worry about the mutable nature of the class and equality as that interacts with uses of hashability.

```suggestion
                        existing_peers = {c.peer_node_id: timestamp for c, timestamp in new_peers_with_peak}
                        peers = self.get_peers_with_peak(peak_hash)
                        new_peers_with_peak = [(c, existing_peers.get(c.peer_node_id, end)) for c in peers]
```

Reply:
oh, I see